dexlib2/src/main/java/com/android/tools/smali/util/ArraySortedSet.java

```diff
@@ -51,10 +49,20 @@ private ArraySortedSet(@Nonnull Comparator<? super T> comparator, @Nonnull T[] a
         this.arr = arr;
     }
 
+    private ArraySortedSet(@Nonnull Comparator<? super T> comparator, @Nonnull Collection<? extends T> collection) {
+        // we assume collection is already sorted by comparator, and all entries are unique
+        this.comparator = comparator;
+        this.arr = collection.toArray();
```

Comment:
We could consider adding a loop to assert this. If running with assertions enables clients should expect longer running time caused by expensive asserts.

Maybe add
```
private boolean assertSorted() {
  for (int i = 1; i < arr.length; i++) {
    assert comparator.compare(arr[i - 1], arr[i]) < 0;
  }
  return true;
}
```
and add `assert assertSorted()` here.

Reply:
Done, and added some testing for this class too.